compiler/z/codegen/S390Peephole.cpp

```diff
@@ -0,0 +1,3248 @@
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#include "z/codegen/S390Peephole.hpp"
+
+#include <stdint.h>                                 // for int32_t, etc
+#include <stdio.h>                                  // for printf, sprintf
+#include <string.h>                                 // for NULL, strcmp, etc
+#include "codegen/CodeGenPhase.hpp"                 // for CodeGenPhase, etc
+#include "codegen/CodeGenerator.hpp"                // for CodeGenerator, etc
+#include "codegen/CodeGenerator_inlines.hpp"
+#include "codegen/ConstantDataSnippet.hpp"
+#include "codegen/FrontEnd.hpp"                     // for TR_FrontEnd, etc
+#include "codegen/InstOpCode.hpp"                   // for InstOpCode, etc
+#include "codegen/Instruction.hpp"                  // for Instruction, etc
+#include "codegen/MemoryReference.hpp"              // for MemoryReference, etc
+#include "codegen/RealRegister.hpp"                 // for RealRegister, etc
+#include "codegen/Register.hpp"                     // for Register
+#include "codegen/RegisterDependency.hpp"
+#include "codegen/RegisterPair.hpp"                 // for RegisterPair
+#include "codegen/Snippet.hpp"                      // for TR::S390Snippet, etc
+#include "ras/Debug.hpp"                            // for TR_DebugBase, etc
+#include "ras/DebugCounter.hpp"
+#include "ras/Delimiter.hpp"                        // for Delimiter
+#include "runtime/Runtime.hpp"                      // for TR_LinkageInfo
+#include "z/codegen/CallSnippet.hpp"
+#include "z/codegen/OpMemToMem.hpp"
+#include "z/codegen/S390Evaluator.hpp"
+#include "z/codegen/S390GenerateInstructions.hpp"
+#include "z/codegen/S390Instruction.hpp"
+#include "z/codegen/S390OutOfLineCodeSection.hpp"
+#include "z/codegen/TRSystemLinkage.hpp"
+
+TR_S390Peephole::TR_S390Peephole(TR::Compilation* comp, TR::CodeGenerator *cg)
+   : _fe(comp->fe()),
+     _outFile(comp->getOutFile()),
+     _cursor(cg->getFirstInstruction()),
+     _cg(cg)
+   {
+   }
+
+TR_S390PreRAPeephole::TR_S390PreRAPeephole(TR::Compilation* comp, TR::CodeGenerator *cg)
+   : TR_S390Peephole(comp, cg)
+   {
+   }
+
+void
+TR_S390PreRAPeephole::perform()
+   {
+   TR::Delimiter d(comp(), comp()->getOption(TR_TraceCG), "preRAPeephole");
+
+   if (comp()->getOption(TR_TraceCG))
+      printInfo("\nPost PreRegister Assignment Peephole Optimization Instructions:\n");
+
+   bool moveInstr;
```

Comment:
I don't see a purpose for the existence of this variable. Are we able to fold this away?

Reply:
f497fd6